src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -608,6 +601,15 @@ public function html(/* $default = null */)
             throw new \InvalidArgumentException('The current node list is empty.');
         }
 
+        if (null !== $this->html5Parser) {
```

Comment:
There is an issue here. You instantiate the HTML5 parser in the constructor even when the content added is not HTML5 but XML or existing DOM elements (coming from elsewhere than a parent crawler using HTML5). This means you might be saving with the HTML5 parser when it was not used for parsing.

Reply:
How do you propose to improve this?

Reply:
well, we need to distinguish 3 cases:

- we are parsing some HTML5
- we are parsing some older HTML
- we are not parsing HTML at all

The boolean argument in the constructor allows us to decide between the first 2 cases at the time we instantiate. But knowing whether this is HTML or no is not something the controller knows (as it can be done later).

The solution might be to store the boolean property. Then, based on that, we would decide which parsing strategy to use *if we load HTML* and instantiate the HTML5 parser if needed.
Then, here, we can keep saying "if I used an HTML5 parser, I also use it for saving".

And for subcrawlers, we copy the content of the private property.